### PR TITLE
Port the list stepped detailed pattern from brochure theme

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Ported the list stepped detailed pattern from the brochure theme with its docs and example.

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/examples/patterns/lists/lists-stepped-detailed/
- Check it looks the same as https://vanilla-framework.github.io/vanilla-brochure-theme/examples/patterns/lists/stepped-list-detailed/
- Check example and docs are ok

## Details
Fixes https://github.com/vanilla-framework/vanilla-framework/issues/1213

### CODE REVIEW

File: docs/en/patterns/lists.md [hunk 1]
```diff
@@ -75,8 +75,20 @@ tutorial or instructions — you can use the class ```.p-list-step```.
     View example of the stepped list pattern
 </a>
 
+## Stepped list: detailed
+
+Stepped list should be used for step by step instructions. This pattern is best
+used on a grey strip as the description sections are displayed in a white box.
```
Comment (barrymcgee) [inline docs/en/patterns/lists.md]: I think this should say 'light strip' as that is the name of the class to use? This confused me doing canonical recently as it was mentioned I should use the grey strip class and I was wondering why `.p-strip--grey` wasn't working.

Comment (kwm14) [inline docs/en/patterns/lists.md]: @barrymcgee +1 to use 'strip--light' for consistency.

**Comment by kwm14** [comment]
Pattern looks good from design.

**Comment by anthonydillon** [comment]
@barrymcgee Done, ready for another look.

**Comment by barrymcgee** [comment]
Can you rebase with develop to sort Percy and also fix conflicts please?

**Pull request merged**